skimage/data/_fetchers.py

```diff
@@ -212,7 +212,7 @@ def _fetch(data_filename):
         return legacy_file_path
 
     # Case 3: file is not present locally
-    if image_fetcher is None:
+    if _image_fetcher is None:
```

Comment:
@lagru we just checked for the same condition (line 198), so shouldn't this block go under the first `if _image_fetcher is None:` conditional?

Reply:
We want to hit case 1 and 2 first, so the content of this block should definitely come after. And `_create_image_fetcher` populates `data_dir` currently in a weird way, requiring us to select the parent directory only if `_image_fetcher is None`. I thought about refactoring this earlier, but I think I wanted to keep the scope of that previous PR small as it was already taking long enough to get the fix in. You could try updating that if you want...

Reply:
I see, thanks for explaining. I'd say it's fine for now.